HighTeach/middleware_login.py

```diff
@@ -0,0 +1,18 @@
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.contrib.auth import authenticate
+
+
+class AdminRedirectMiddleware:
```

Comment:
Why is this here and not as [a view](https://docs.djangoproject.com/en/4.2/topics/auth/default/#limiting-access-to-logged-in-users)?